src/test/system_tests/test_bucket_access.js

```diff
@@ -109,12 +109,33 @@ async function setup() {
 
     let account = account_by_name(system_info.accounts, full_access_user.email);
     full_access_user.access_keys = account.access_keys[0];
+    await Promise.all(
```

Comment:
I am not sure what you are doing here. please add a comment

Reply:
This is again trying to replicate `permission_list`. Instead of manually granting permissions to each bucket, it loops over the `permission_list` and generates S3 policies which will give the user equivalent permissions over the buckets that `permission_list` was giving.